File: src/main/java/uk/gov/justice/hmpps/prison/api/model/QuestionnaireAnswer.java
package uk.gov.justice.hmpps.prison.api.model;

import com.fasterxml.jackson.annotation.JsonInclude;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import org.apache.commons.lang3.builder.CompareToBuilder;

import java.time.LocalDateTime;

@Schema(description = "Questionnaire Answer")
@JsonInclude(JsonInclude.Include.NON_NULL)
@Builder
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(of = {"questionnaireAnsId"})
@Data
public class QuestionnaireAnswer implements Comparable<QuestionnaireAnswer> {

    @Schema(required = true, description = "ID for this Answer")
    private Long questionnaireAnsId;
    @Schema(required = true, description = "", example = "1")
    private int answerSeq;
    @Schema(required = true, description = "", example = "YES")
    private String answerDesc;
    @Schema(required = true, description = "", example = "1")
    private int answerListSeq;
    @Schema(required = true, description = "", example = "true")
    private Boolean answerActiveFlag;
    @Schema(required = true, description = "", example = "2017-01-02T00:00:00")
    private LocalDateTime answerExpiryDate;
    @Schema(required = true, description = "", example = "false")
    private Boolean dateRequiredFlag;
    @Schema(required = true, description = "", example = "false")
    private Boolean commentRequiredFlag;

    @Override
    public int compareTo(final QuestionnaireAnswer answer) {
        return new CompareToBuilder()
                .append(this.getAnswerSeq(), answer.getAnswerSeq())
                .toComparison();
    }
}

File: src/main/java/uk/gov/justice/hmpps/prison/api/model/QuestionnaireAnswer.kt
package uk.gov.justice.hmpps.prison.api.model

import com.fasterxml.jackson.annotation.JsonInclude
import io.swagger.v3.oas.annotations.media.Schema
import io.swagger.v3.oas.annotations.media.Schema.RequiredMode
import java.time.LocalDate

@Schema(description = "Questionnaire Answer")
@JsonInclude(JsonInclude.Include.NON_NULL)
data class QuestionnaireAnswer(
  @Schema(requiredMode = RequiredMode.REQUIRED, description = "ID for this Answer")
  private val questionnaireAnsId: Long,

  @Schema(requiredMode = RequiredMode.REQUIRED, description = "", example = "1")
  private val answerSeq: Int = 0,

  @Schema(requiredMode = RequiredMode.REQUIRED, description = "Answer Text", example = "YES")
  private val answerDesc: String,

  @Schema(requiredMode = RequiredMode.REQUIRED, description = "Answer Sequence", example = "1")
  private val answerListSeq: Int = 0,

  @Schema(requiredMode = RequiredMode.REQUIRED, description = "", example = "true")
  private val answerActiveFlag: Boolean = true,

  @Schema(requiredMode = RequiredMode.REQUIRED, description = "", example = "2017-01-02T00:00:00")
  private val answerExpiryDate: LocalDate? = null,

  @Schema(
    requiredMode = RequiredMode.REQUIRED,
    description = "Should the answer include date information?",
    example = "false",
  )
  private val dateRequiredFlag: Boolean = false,

  @Schema(requiredMode = RequiredMode.REQUIRED, description = "Should the answer include Comment?", example = "false")
  private val commentRequiredFlag: Boolean = false,

  @Schema(requiredMode = RequiredMode.REQUIRED, description = "Next questionnaire question ID")
  private val nextQuestionnaireQueId: Long? = null,

)
